Content.Server/Physics/Controllers/MoverController.cs

```diff
@@ -77,23 +77,26 @@ public override void UpdateBeforeSolve(bool prediction, float frameTime)
                     continue;
                 }
 
-                PhysicsComponent? body;
+                var mob = uid;
+                PhysicsComponent? mobBody;
+
                 var xformMover = xform;
 
                 if (mover.ToParent && relayQuery.HasComponent(xform.ParentUid))
                 {
-                    if (!bodyQuery.TryGetComponent(xform.ParentUid, out body) ||
-                        !TryComp(xform.ParentUid, out xformMover))
+                    mob = xform.ParentUid;
+                    if (!bodyQuery.TryGetComponent(mob, out mobBody) ||
+                        !TryComp(mob, out xformMover))
                     {
                         continue;
                     }
                 }
-                else if (!bodyQuery.TryGetComponent(uid, out body))
+                else if (!bodyQuery.TryGetComponent(uid, out mobBody))
                 {
                     continue;
                 }
-
-                HandleMobMovement(uid, mover, body, xformMover, frameTime, xformQuery, moverQuery, relayTargetQuery);
+                var speedMod = CompOrNull<MovementSpeedModifierComponent>(uid);
+                HandleMobMovement(mob, mover, mobBody, xformMover, frameTime, xformQuery, moverQuery, relayTargetQuery, speedMod);
```

Comment:
All of these changes look pointless

Reply:
i changed this because of assertion error - body might not be the component of uid (if there is relay)